### PR TITLE
[v7r1] Modifications to manage certificates to access the Elias (Elaasticsea…

### PR DESCRIPTION
**Pull request by sanguillon**
Configuration we used in /opt/dirac/etc/dirac.cfg to access the ELIAS database hosted at CC-IN2P3
-----------------------------
/DIRAC/Setups/CTA-cert/Monitoring = Certification

/Systems/Monitoring/Certification/Databases/MonitoringDB
          {
          Host = elias-beta.cc.in2p3.fr
          Port = 9200
          SSL = False
          IndexPrefix = cta-
          CRT = True
          ca_certs = /opt/dirac/etc/grid-security/ES/ca.crt
          client_key = /opt/dirac/etc/grid-security/ES/ctacertbot.key
          client_cert = /opt/dirac/etc/grid-security//ES/ctacertbot.crt
        }

BEGINRELEASENOTES

*Monitoring
NEW: added possibility to login to ES via certificates.

ENDRELEASENOTES

### CODE REVIEW

File: ConfigurationSystem/Client/Utilities.py [hunk 1]
```diff
@@ -530,6 +530,62 @@ def getElasticDBParameters(fullname):
     ssl = False if result['Value'].lower() in ('false', 'no', 'n') else True
   parameters['SSL'] = ssl
 
+  # Elasticsearch use certs
+  result = gConfig.getOption(cs_path + '/CRT')
+  if not result['OK']:
+    # No host name found, try at the common place
```
Comment (fstagni) [inline ConfigurationSystem/Client/Utilities.py:536]: Wrong (copy-pasted) comment (also below).

File: ConfigurationSystem/Client/Utilities.py [hunk 2]
```diff
@@ -530,6 +530,62 @@ def getElasticDBParameters(fullname):
...
+    result = gConfig.getOption('/Systems/NoSQLDatabases/CRT')
+    if not result['OK']:
+      gLogger.warn("Failed to get the configuration parameter: certs. Using False")
```
Comment (fstagni) [inline ConfigurationSystem/Client/Utilities.py:539]: `certs` or `CRT` ?

File: Core/Utilities/ElasticSearchDB.py [hunk 3]
```diff
@@ -105,6 +110,14 @@ def __init__(self, host, port, user=None, password=None, indexPrefix='', useSSL=
                                   use_ssl=True,
                                   verify_certs=True,
                                   ca_certs=casFile)
+    elif useCRT:
+        self.client = Elasticsearch(self.__url,
+                                  timeout=self.__timeout,
```
Comment (fstagni) [inline Core/Utilities/ElasticSearchDB.py:115]: bad indentation.

**Review by fstagni — CHANGES_REQUESTED** [review]
Thanks!
It's missing some changes in the documentation too.

**Comment by sanguillon** [comment]
Hi Federico,

I am sorry for these errors that I fixed.
Could you please tell me the procedure I have to use to check/validate the code before submitting a pull request?
Thank you in advance for your answer.

Michele

**Comment by fstagni** [comment]
No problem, we all commit/fix/re-fix PRs. There's quite some doc in https://dirac.readthedocs.io/en/integration/DeveloperGuide/index.html and for the tests we run a mix of unit and integration tests, as well as static code checks and so on. 
We also run "certification hackathons" every 2nd week on pre-releases.

File: ConfigurationSystem/Client/Utilities.py (hunk 1 above)
Comment (fstagni) [reply ConfigurationSystem/Client/Utilities.py:536]: Just change this comment please, and the ones below, and then I approve.

**Comment by sanguillon** [comment]
Done! Thank you!
Cheers,
Michèle

**Review by fstagni — APPROVED** [review]

**Pull request merged**